src/github.com/ebay/beam/rpc/kgobject.go

```diff
@@ -43,7 +43,8 @@ type KGObject struct {
 	//  LangID:		19 bytes [only for string]
 	//
 	// TypeBytes
-	//	String: 	the utf8 bytes of the string, no terminator, length is not encoded anywhere
+	//  String:     The utf8 bytes of the string, a terminating 0x00. length is not encoded anywhere.
+	//              The nil is to ensure correct ordering, its not used to determine the end of the string.
```

Comment:
Please update the comment to include whether the string must contain valid UTF-8 (I think the answer is no) and whether it may contain null bytes (I think the answer is yes).